lib/icinga/icingaapplication.cpp

```diff
@@ -291,6 +294,33 @@ String IcingaApplication::GetNodeName(void) const
 	return ScriptGlobal::Get("NodeName");
 }
 
+void IcingaApplication::DeclarePluginDir(const String& pluginDir)
+{
+	if (!ScriptGlobal::Exists("PluginDir"))
+		return ScriptGlobal::Set("PluginDir", pluginDir);
+}
+
+String IcingaApplication::GetPluginDir(void)
+{
+	return ScriptGlobal::Get("PluginDir");
+}
+
+void IcingaApplication::DeclarePluginPath(const String& pluginPath)
+{
+	if (!ScriptGlobal::Exists("PluginPath"))
+		return ScriptGlobal::Set("PluginPath", Array::FromVector(String(pluginPath).Split(":")));
+}
+
+Value IcingaApplication::GetPluginPath(void)
+{
+	Array::Ptr path = ScriptGlobal::Get("PluginPath");
+	String dir = GetPluginDir();
+	if (!dir.IsEmpty() && !path->Contains(dir))
+		path->Add(dir);
```

Comment:
Not a super-huge fan of this but I don't really have a better idea (for now anyway).